### PR TITLE
Add `nuget` package provider

### PR DESCRIPTION
**Pull request by compnerd**
Introduce the `nuget` package provider, which is the only option on
Windows, but is portable to other systems as well. This is required
for enabling the test suite on Windows.

Refs: #4369

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: Sources/PackageLoading/Target+PkgConfig.swift [hunk 1]
```diff
@@ -148,6 +150,13 @@ extension SystemPackageProviderDescription {
             if case .linux(.fedora) = platform {
                 return true
             }
+        case .nuget:
+            switch platform {
+            case .darwin, .windows, .linux:
```
Comment (tomerd) [inline Sources/PackageLoading/Target+PkgConfig.swift:155]: should this be windows only? is there formal support and practical adoption of nuget on linux and darwin?

Comment (compnerd) [reply Sources/PackageLoading/Target+PkgConfig.swift:155]: Yes, they are formally supported and adopted - even though the primary usage is .NET, but that is more a historical artifact.

**Review by tomerd — APPROVED** [review]

**Comment by compnerd** [comment]
Seems that it doesn't like the 9999 version, need to figure out what version main is.

File: Sources/PackageLoading/Target+PkgConfig.swift (hunk 1 above)
Comment (compnerd) [reply Sources/PackageLoading/Target+PkgConfig.swift:155]: A nice example package: https://www.nuget.org/packages/runtime.debian.8-x64.runtime.native.System.Security.Cryptography.OpenSsl/

Prebuilt .so for Debian x86_64.

**Comment by tomerd** [comment]
@abertelrud should know. it may be the case that we need to make a change on `main` to enable it again

**Comment by abertelrud** [comment]
> @abertelrud should know. it may be the case that we need to make a change on `main` to enable it again

I think it's just 999.

File: Sources/PackageDescription/PackageDescription.swift [hunk 2]
```diff
@@ -388,6 +388,8 @@ public enum SystemPackageProvider {
     /// Packages installable by the yum package manager.
     @available(_PackageDescription, introduced: 5.3)
     case yumItem([String])
+    @available(_PackageDescription, introduced: 9999)
```
Comment (abertelrud) [inline Sources/PackageDescription/PackageDescription.swift:391]:
```suggestion
    @available(_PackageDescription, introduced: 999.0)
```

File: Sources/PackageDescription/PackageDescription.swift [hunk 3]
```diff
@@ -420,6 +422,17 @@ public enum SystemPackageProvider {
...
         return .yumItem(packages)
     }
+
+    /// Creates a system package provider with a list of installable packages
+    /// for users of the nuget package manager on Linux or Windows.
+    ///
+    /// - Parameter packages: The list of package names.
+    ///
+    /// - Returns: A package provider.
+    @available(_PackageDescription, introduced: 9999)
```
Comment (abertelrud) [inline Sources/PackageDescription/PackageDescription.swift:432]:
```suggestion
    @available(_PackageDescription, introduced: 999.0)
```

**Review by abertelrud — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**